app/src/main/java/com/connorlay/knitgrid/adapters/PatternLegendRecyclerViewAdapter.java

```diff
@@ -49,16 +47,13 @@ public void onBindViewHolder(StitchRecycleViewHolder holder, int position) {
         final Stitch stitch = mStitches.get(position);
         holder.textView.setText(stitch.getAbbreviation());
 
+        ImageView view = holder.imageView;
+        view.setImageResource(stitch.getIconID());
 
-        Context context = holder.imageView.getContext();
-
-        Picasso.with(context).load(stitch.getIconID()).fit().centerInside().into(holder.imageView);
         holder.fullView.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (mListener != null) {
-                    mListener.onStitchRowClick(stitch);
-                }
+                if (mListener != null) mListener.onStitchRowClick(stitch);
```

Comment:
So what Connor was actually getting at was this:

`if (mListener == null) { return; }`
`mListener.onStitchRowClick(stitch);`

Either change it to this, or go back to the original. Either is better than this one-line if statement.